x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/users/observed_details/query.observed_user_details.dsl.ts

```diff
@@ -28,6 +30,10 @@ export const buildUserDetailsQuery = ({
     },
   ];
 
+  if (experimentalFeatures?.newUserDetailsFlyout) {
+    filter.push(NOT_EVENT_KIND_ASSET_FILTER);
```

Comment:
On the new flyout, we must filter out by `event.kind='asset'` to avoid displaying the Azure data on the Observed data table.

Reply:
Make sens, could we add this case in its units test please?

Reply:
Another idea, I found in `x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/users/kpi/total_users/query.build_total_users_kpi.dsl.ts` we just pass `filterQuery` to build the query. If we do that here, we might not need to pass `experimentalFeatures ` as the 2nd param here.